solidity/contracts/isms/ccip-read/CommitmentReadIsm.sol

```diff
@@ -54,30 +54,34 @@ contract CommitmentReadIsm is AbstractCcipReadIsm, Ownable {
 
     /**
      * @notice Verifies the commitment by comparing the calldata hash to the commitment
-     * @param _metadata The encoded (salt, calls) whose hash is the commitment
+     * @param _metadata The encoded (ica, salt, calls)
      * @param _message The reveal hyperlane message
      * @return true If the hash of the metadata matches the commitment.
      */
     function verify(
         bytes calldata _metadata,
         bytes calldata _message
     ) external returns (bool) {
-        (CallLib.Call[] memory calls, bytes32 salt, OwnableMulticall ica) = abi
-            .decode(_metadata, (CallLib.Call[], bytes32, OwnableMulticall));
+        // Fetch encoded ica, salt, and calls
+        address _ica = address(bytes20(_metadata[:20]));
+        OwnableMulticall ica = OwnableMulticall(payable(_ica));
 
-        bytes32 revealedHash = keccak256(_metadata);
+        bytes32 salt = bytes32(_metadata[20:52]);
+
+        CallLib.Call[] memory calls = abi.decode(
+            _metadata[52:],
+            (CallLib.Call[])
+        );
```

Comment:
you can move the decoding to below the commitment check

Reply:
should we use a library similar to InterchainAccountMessage for this?

Reply:
dont love these "magic constants"

Reply:
maybe we can add a lint rule for this

Reply:
A library sounds good, I'll follow up